### PR TITLE
Er/callback completion test

### PR DESCRIPTION
**Pull request by msimko81**
Added tests for polling and callback functionalities.

### CODE REVIEW

**Review by hradilm — APPROVED** [review]
I have checked the code and run the migration worker on this branch to see if it did not broke anything. Seems to be ok. But it would be wonderfull if @erymski or @vasicem would review it as well before merging.

File: WebApplication.Tests/PublisherTest.cs [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+        public PublisherTest()
+        {
+            _workItemArgs = new Dictionary<string, IArgument>();
+
+            _configMock = new Mock<IForgeAppBase>();
+            _configMock.Setup(mock => mock.ActivityId).Returns("activityId");
+            _configMock.Setup(mock => mock.ActivityLabel).Returns("activityLabel");
+        }
+        
+        private class CallbackDataProvider : IEnumerable<object[]>
```
Comment (mivasi) [inline WebApplication.Tests/PublisherTest.cs:32]: I know we maybe already talked about it on another test. But why not to use simple Setup and we are using Theory and DataProvider?

File: WebApplication/Processing/Publisher.cs [hunk 2]
```diff
@@ -102,12 +112,12 @@ private async Task<WorkItemStatus> LaunchAndWait(WorkItem wi)
...
         private async Task<WorkItemStatus> RunWithPolling(WorkItem wi)
         {
-            WorkItemStatus status = await _client.CreateWorkItemAsync(wi);
+            WorkItemStatus status = (await _workItemsApi.CreateWorkItemAsync(wi)).Content;
             _logger.LogInformation($"Created WI {status.Id} (polling mode)");
             while (status.Status == Status.Pending || status.Status == Status.Inprogress)
             {
-                await Task.Delay(2000);
-                status = await _client.GetWorkitemStatusAsync(status.Id);
+                _taskUtil.Sleep(2000);
```
Comment (mivasi) [inline WebApplication/Processing/Publisher.cs:119]: You are missing await here.

**Review by mivasi — CHANGES_REQUESTED** [review]

File: WebApplication.Tests/PublisherTest.cs (hunk 1 above)
Comment (msimko81) [reply WebApplication.Tests/PublisherTest.cs:32]: There are two different tests for callbacks. 

Moreover, in the next commit I have refactored the test and join callback and polling to the single Theory with three different sets of test data. Two for callback, one for polling.

**Review by mivasi — APPROVED** [review]

**Pull request merged**